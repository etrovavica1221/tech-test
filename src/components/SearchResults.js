import React from 'react';
import '../styles/SearchResults.css';
import PropTypes from 'prop-types';

const SearchResults = ({ results }) => {
    if (!results.length) {
        return <p id="p-NoResults">No results</p>   
    } else {  
        return(
            <span>
                <div id="p-SearchResults">
                    <p>Search Results:</p>
                </div>
                <div className="SearchResults" data-testid="resultTest">
                        {results.map((image) => (
                            <img className="card-image" src={image} alt="spaceImage" key={image} />
                        ))}
                </div>
            </span>
        );
    }
};

SearchResults.propTypes = {
    image: PropTypes.func,
    results: PropTypes.array,
}

export default SearchResults;
