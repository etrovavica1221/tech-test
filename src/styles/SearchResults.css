.SearchResults {
    display: flex;
    flex-direction: column;
    margin: auto;
    border-style: ridge;
    padding: 20px;
    position: relative;
    top: 30px;
}

p {
    margin: 30px auto;
    font-size: 25px;
}