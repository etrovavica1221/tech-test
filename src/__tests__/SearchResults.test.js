import React from 'react';
import { render } from "@testing-library/react";
import SearchResults from '../components/SearchResults';

describe("SearchResults", () => {
    const results = [
        "https://images-assets.nasa.gov/image/GSFC_20171208â€¦e_e001517/GSFC_20171208_Archive_e001517~thumb.jpg",
        "https://images-assets.nasa.gov/image/PIA09322/PIA09322~thumb.jpg"
    ]; 
 
    it('renders correctly', () => {
        const { asFragment } = render(<SearchResults results={results}/>);
        expect(asFragment()).toMatchSnapshot();
    })

    it('renders the correct amount of SearchResults components', () => {
        const { getByTestId } = render(<SearchResults results={results} />)

        expect(getByTestId("resultTest")).toHaveClass("SearchResults");
    })
})
