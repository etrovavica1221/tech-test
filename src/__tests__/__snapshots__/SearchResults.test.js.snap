// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <span>
    <div
      id="p-SearchResults"
    >
      <p>
        Search Results:
      </p>
    </div>
    <div
      class="SearchResults"
      data-testid="resultTest"
    >
      <img
        alt="spaceImage"
        class="card-image"
        src="https://images-assets.nasa.gov/image/GSFC_20171208â€¦e_e001517/GSFC_20171208_Archive_e001517~thumb.jpg"
      />
      <img
        alt="spaceImage"
        class="card-image"
        src="https://images-assets.nasa.gov/image/PIA09322/PIA09322~thumb.jpg"
      />
    </div>
  </span>
</DocumentFragment>
`;
